sdk/src/transaction_context.rs

```diff
@@ -65,16 +70,20 @@ impl TransactionContext {
                 .into_iter()
                 .map(|(key, account)| (key, RefCell::new(account)))
                 .unzip();
+        let account_touched_flags = vec![false; accounts.len()];
         Self {
             account_keys: Pin::new(account_keys.into_boxed_slice()),
             accounts: Pin::new(accounts.into_boxed_slice()),
+            account_touched_flags: RefCell::new(Pin::new(account_touched_flags.into_boxed_slice())),
             instruction_context_capacity,
             instruction_stack: Vec::with_capacity(instruction_context_capacity),
             number_of_instructions_at_transaction_level,
             instruction_trace: Vec::with_capacity(number_of_instructions_at_transaction_level),
             return_data: TransactionReturnData::default(),
             total_resize_limit,
             total_resize_delta: RefCell::new(0),
+            enable_early_verification_of_account_modifications: false,
+            rent: Rent::default(),
```

Comment:
I'd like to see these new `::default()` initializations go away if possible. I'm a bit concerned that we accidentally introduce a code path where `enable_early_verification_of_account_modifications()` won't be called, resulting in wrong or missed runtime checks. obviously, it's not currently not existing.

seems like changing the signature of `TransactionContext::new()` isn't not that hard, right? you can just explicitly pass defaults to reduce plumbing from non-critical code pathes like bench.